File: 028__Implement_strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if needle == "":
            return 0
        m, n = len(haystack), len(needle)
        if haystack == needle :
            return 0

        for i in range(m - n + 1):
            if haystack[i:i+n] == needle:
                return i
        return -1
        

File: Java/028__Implement_strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack.length() < needle.length()) {
            return -1;
        }
        if (haystack.length() == needle.length()) {
            return haystack.equals(needle) ? 0 : -1;
        }
        int i = 0, j = 0;
        while (i <= haystack.length() - needle.length()) {
            boolean check = true;
            for (j = 0; j < needle.length(); j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    check = false;
                }
            }
            if (check) {
                return i;
            }
            i++;
        }
        return -1;
    }
}
